src/traces/heatmap/has_columns.js

```diff
@@ -6,9 +6,10 @@
 * LICENSE file in the root directory of this source tree.
 */
 
-
 'use strict';
 
+var Lib = require('../../lib');
+
 module.exports = function(trace) {
-    return !Array.isArray(trace.z[0]);
+    return !Lib.isArrayOrTypedArray(trace.z[0]);
```

Comment:
Would it work to provide a 2D array as an array of typed arrays? Not that I want to encourage this, far better for us to provide a solution based on a single typed array...

Reply:
> Would it work to provide a 2D array as an array of typed arrays?

Should have looked at the [next commit](https://github.com/plotly/plotly.js/pull/2388/commits/09d37b670b013e71b607bd99ee010b1c4ddff02e#diff-ccbe72dad26409c70d2a8001ee8134ffR453) before commenting 🏆

Reply:
Yep, arrays of typed arrays for should work after this PR.